src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/haflow/DeallocateResourcesAction.java

```diff
@@ -44,8 +44,8 @@ public void perform(S from, S to, E event, C context, T stateMachine) {
         stateMachine.getOldResources().forEach(haFlowResources -> {
             transactionManager.doInTransaction(() ->
                     resourcesManager.deallocateHaFlowResources(haFlowResources));
-            HaFlowHistoryService.using(stateMachine.getCarrier()).save(HaFlowHistory
-                    .withTaskId(stateMachine.getCommandContext().getCorrelationId())
+            FlowHistoryService.using(stateMachine.getCarrier()).save(HaFlowHistory
```

Comment:
May we not create a new instance of FlowHistoryService each iteration?

Reply:
I created a local variable for the flow history service.